Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -626,8 +640,11 @@ bool RenderTheme::paint(const RenderBox& box, ControlStates& controlStates, cons
     
     auto type = box.style().effectiveAppearance();
 
-    if (UNLIKELY(!canPaint(paintInfo, box.settings(), type)))
+    if (UNLIKELY(!canPaint(paintInfo, box.settings(), type))) {
+        if (type == ControlPartType::MenulistButton || type == ControlPartType::TextField || type == ControlPartType::TextArea || type == ControlPartType::Listbox)
```

Comment:
Why is this change not being made in the `canPaint` method? Also, why are unrelated types being added in this patch?

Reply:
Fixed. Check was moved to canPaint().